lightning-invoice/src/utils.rs

```diff
@@ -627,7 +627,7 @@ where
 	log_trace!(logger, "Considering {} channels for invoice route hints", channels.len());
 	for channel in channels.into_iter().filter(|chan| chan.is_channel_ready) {
 		if channel.get_inbound_payment_scid().is_none() || channel.counterparty.forwarding_info.is_none() {
-			log_trace!(logger, "Ignoring channel {} for invoice route hints", log_bytes!(channel.channel_id));
+			log_trace!(WithChannelDetails{ logger, channel_details: &channel }, "Ignoring channel {} for invoice route hints", log_bytes!(channel.channel_id));
```

Comment:
Would be better to wrap the logger once at the start of the method.

```rust
let logger = WithChannelDetails { logger, channel_details: &channel };
```

That way the calls sites don't need to change.

Reply:
Err... at least within the loop on this one. 😅